src/ScottPlot5/ScottPlot/Plottables/Heatmap.cs

```diff
@@ -0,0 +1,123 @@
+﻿using ScottPlot.Axis;
+using SkiaSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using ScottPlot.Colormaps;
+
+namespace ScottPlot.Plottables
+{
+    public enum CellAlignment
+    {
+        Start,
+        Center,
+        End
+    }
+
+    public class Heatmap : IPlottable
+    {
+        public bool IsVisible { get; set; } = true;
+        public IAxes Axes { get; set; } = Axis.Axes.Default;
+        public IColormap Colormap { get; set; } = new Viridis();
+        public CellAlignment CellAlignment { get; set; } = CellAlignment.Center;
+        public CoordinateRect? Extent { get; set; }
+        public bool FlipVertically { get; set; } = false;
+        public bool Smooth { get; set; } = false;
+
+        private CoordinateRect ExtentOrDefault => Extent ?? new CoordinateRect(0, Intensities.GetLength(0), 0, Intensities.GetLength(1));
+        private CoordinateRect AlignedExtent => ExtentOrDefault.WithTranslation(new(CellAlignmentFraction * CellWidth, CellAlignmentFraction * CellHeight));
+        private double CellAlignmentFraction => CellAlignment switch
+        {
+            CellAlignment.Start => 0,
+            CellAlignment.Center => -0.5,
+            CellAlignment.End => -1,
+            _ => throw new NotImplementedException(),
+        };
+        private double CellWidth => ExtentOrDefault.Width / Intensities.GetLength(0);
+        private double CellHeight => ExtentOrDefault.Height / Intensities.GetLength(1);
+
+
+        private double[,] _intensities;
+        public double[,] Intensities
+        {
+            get => _intensities;
+            set
+            {
+                _intensities = value;
+                Update();
+            }
+        }
+
+        private SKBitmap? bitmap = null;
+
+        public Heatmap(double[,] intensities)
```

Comment:
Let's do what we have to so we kill all the warnings.

I may enable `TreatWarningsAsErrors` in the CI pipeline to help with this 😈

Reply:
There were only nullability warnings because `Intensities` had a non-trivial setter, so the compiler couldn't tell that it set the backing field. I ended up removing the setter on `Intensities` anyways, but in the future this might be the sort of place where I would explicitly disable the warning, as it was incorrect, the backing field was definitively set to non-null (provided the constructor receives a non-null value).